src/bin/ion/commands/generate/templates/typescript/nested_type.templ

```diff
@@ -0,0 +1,104 @@
+{% import "util_macros.templ" as macros %}
+
+{% if model.annotations %}
+{%- for annotation in model.annotations %}
+@{{ annotation }}
+{%- endfor %}
+{% endif %}
+export interface {{ model.name }} extends IonSerializable {
+    {% for field in model.fields %}
+    {% if field.annotations %}
+    {%- for annotation in field.annotations %}
+    @{{ annotation }}
+    {%- endfor %}
+    {% endif %}
+    {{ field.name }}: {{ field | nullable_type }};
+    {% endfor %}
+}
+
+{{ macros::type_guard(model=model) }}
+
+export class {{ model.name }}Impl implements {{ model.name }} {
```

Comment:
Similar to other templates maybe you can just create a macro for the class in `interface.templ` and use that macro in this template so that you dont have to write the same template again.